### DISCUSSION TITLE
Why Symfony 5 does not use the latest version of bundles?

### DISCUSSION THREAD

**Discussion by karaka200** [symfony/symfony#42523] [Q&A] [answered]
Why Symfony 5 does not use the latest version of bundles?

For example:

```
"symfony/flex": "1.3.1" [But the latest version is 1.13.4]
"doctrine/annotations": "1.13" [But the latest version is 1.13.2]
"doctrine/doctrine-bundle": "2.4" [But the latest version is 2.4.2]
"doctrine/doctrine-migrations-bundle": "3.1" [But the latest version is 3.1.1]
"doctrine/orm": "2.9" [But the latest version is 2.9.4]
"twig/twig": "2.12|3.0" [But the latest version is 3.3.2]
...

```

And when I use composer update, show this:

```
Loading composer repositories with package information
Restricting packages listed in "symfony/symfony" to "5.3.*"
Updating dependencies
Nothing to modify in lock file
Installing dependencies from lock file (including require-dev)
Nothing to install, update or remove
Generating optimized autoload files
composer/package-versions-deprecated: Generating version class...
composer/package-versions-deprecated: ...done generating version class
68 packages you are using are looking for funding.
Use the `composer fund` command to find out more!

Run composer recipes at any time to see the status of your Symfony recipes.

Executing script cache:clear [OK]
Executing script assets:install public [OK]

```

I use PHP 8

**Comment by linaori** [comment, marked as answer]
is the first one you're showing from your composer.json? If so, then you're looking in the wrong file as the composer.lock contains the installed versions.

You can verify your installed packages with `composer show --outdated`

**Reply by karaka200** [reply]
So what are the versions in composer.json?

**Reply by faizanakram99** [reply]
This has nothing to do with Symfony btw. I suggest you to read documentation of `composer` at https://getcomposer.org

**Reply by karaka200** [reply]
When I use `composer show --outdated `, show this:

```
Color legend:
- patch or minor release available - update recommended
- major release available - update possible
doctrine/dbal 2.13.2 3.1.1 Powerful PHP database abstraction layer (DBAL) with many features for database schema in...
psr/cache     2.0.0  3.0.0 Common interface for caching libraries
psr/container 1.1.1  2.0.1 Common Container Interface (PHP FIG PSR-11)
psr/log       2.0.0  3.0.0 Common interface for logging libraries
```

but I can't update them.

**Reply by ahundiak** [reply]
It is mildly interesting that I get the same results on a freshly create Symfony 5.3 app.  Notice that the newer version is in yellow.  That means that a newer version is available but it could involve breaking changes.  In particular, there is a relatively new Doctrine DBAL version 3 out which is a significant rewrite of version 2.  However, version 3 for Doctrine ORM is not yet released.

Bottom line is that you can ignore yellow.  The ones highlighted in red are the ones that need updating.

I find this [composer article](https://andy-carter.com/blog/six-lesser-known-composer-commands-you-should-know) to be handy and mostly accurate.

**Reply by zspine** [reply]
You could try ```composer outdated --direct```
Only your direct dependencies are taken into consideration when using composer outdated --direct. 

red: Important and can be updated without making any changes to your code base
yellow: Breaking changes and may require your application code to be updated
